### PR TITLE
feat: add '--server-addr' in sqlness runner

### PR DESCRIPTION
**Pull request by zyy17**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Add `--server-addr` in sqlness runner, we can run the tests like:

```console
cargo sqlness --server-addr 127.0.0.1:14001 --test-filter standalone

cargo sqlness --server-addr 127.0.0.1:24001 --test-filter distributed
```

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
closes https://github.com/GreptimeTeam/greptimedb/issues/2635

### CODE REVIEW

**Comment by zyy17** [comment]
It seems the `--mode` is duplicated with `--test-filter`. I'll remove it later.

File: tests/runner/src/env.rs [hunk 1]
```diff
@@ -52,8 +54,46 @@ impl EnvController for Env {
...
+                        server_processes: None,
+                        metasrv_process: None,
+                        frontend_process: None,
+                        ctx: GreptimeDBContext {
+                            time: 0,
+                            datanode_id: Default::default(),
+                        },
+                        is_standalone: false,
+                        env: self.clone(),
+                    }
```
Comment (killme2008) [inline tests/runner/src/env.rs:72]: Can we refactor the function `start_standalone` and `start_distributed` to make them accept `&Option<String>` which represents the optional server address? Those logics look the same but with a few modifications.

Comment (killme2008) [reply tests/runner/src/env.rs:72]: If so, we can just change this into:

```rust
self.start_standalone(&start_distributed).await;
```

Comment (zyy17) [reply tests/runner/src/env.rs]: The `Env` struct already has the `server_addr` field, and I put the duplicated logic in a new function `connect_db()`.

**Review by killme2008 — APPROVED** [review]
LGTM. @waynexia PTAL

**Review by waynexia — APPROVED** [review]

**Pull request merged**